src/test/cypress/integration/exam/ExamParticipation.spec.ts

```diff
@@ -0,0 +1,122 @@
+import { GET, BASE_API } from '../../support/constants';
+import { CypressExamBuilder } from '../../support/requests/CourseManagementRequests';
+import { artemis } from '../../support/ArtemisTesting';
+import { generateUUID } from '../../support/utils';
+import dayjs from 'dayjs';
+import submission from '../../fixtures/programming_exercise_submissions/all_successful/submission.json';
+
+// Requests
+const courseRequests = artemis.requests.courseManagement;
+
+// User management
+const users = artemis.users;
+const student = users.getStudentOne();
+
+// Pageobjects
+const courses = artemis.pageobjects.courses;
+const courseOverview = artemis.pageobjects.courseOverview;
+const examStartEnd = artemis.pageobjects.examStartEnd;
+const examNavigation = artemis.pageobjects.examNavigationBar;
+const onlineEditor = artemis.pageobjects.onlineEditor;
+
+// Common primitives
+const uid = generateUUID();
+const examTitle = 'exam' + uid;
+const submissionText = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.';
+const textExerciseTitle = 'Cypress text exercise';
+
+describe('Exam participation', () => {
+    let course: any;
+    let exam: any;
+
+    before(() => {
+        cy.login(users.getAdmin());
+        courseRequests.createCourse().then((response) => {
+            course = response.body;
+            const examContent = new CypressExamBuilder(course)
+                .title(examTitle)
+                .visibleDate(dayjs().subtract(3, 'days'))
+                .startDate(dayjs().subtract(2, 'days'))
+                .endDate(dayjs().add(3, 'days'))
+                .maxPoints(20)
+                .numberOfExercises(2)
+                .build();
+            courseRequests.createExam(examContent).then((examResponse) => {
+                exam = examResponse.body;
+                courseRequests.registerStudentForExam(exam, student);
+                courseRequests.addExerciseGroupForExam(exam).then((groupResponse) => {
+                    courseRequests.createAndAddTextExerciseToExam(groupResponse.body, textExerciseTitle);
+                });
+                courseRequests.addExerciseGroupForExam(exam).then((groupResponse) => {
```

Comment:
Is there no problem if all exercise groups have the same name? Honest question this just surprises me.

Reply:
The default value generates a uuid, so it's not always the same name